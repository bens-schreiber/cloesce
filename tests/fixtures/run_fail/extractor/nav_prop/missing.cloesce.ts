// @ts-nocheck

@D1
class Foo {
  @PrimaryKey
  id: number;

  @OneToOne() // missing generic
  bar: Bar;
}
