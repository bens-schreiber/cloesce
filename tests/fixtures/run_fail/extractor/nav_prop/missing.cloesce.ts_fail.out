==== CLOESCE ERROR ====
Error [MissingNavigationPropertyReference]: Missing Navigation Property Reference
Phase: TypeScript IDL Extraction
Context: Foo.bar
@OneToOne() // missing generic
  bar: Bar;

Suggested fix: Navigation properties require a foreign key model attribute reference
